src/jsonexpr/evaluator.ts

```diff
@@ -1,12 +1,16 @@
+/* eslint-disable */
 import { isEqualsDeep, isObject } from "../utils";
 
 export class Evaluator {
-	constructor(operators, vars) {
+	private readonly operators: any;
+	private readonly vars: any;
+
+	constructor(operators: any, vars: any) {
 		this.operators = operators;
 		this.vars = vars;
 	}
 
-	evaluate(expr) {
+	evaluate(expr: any) {
```

Comment:
Same here: could we make use of generics here as well?

Reply:
Done, although this file is a bit messy type-wise. It works as it is, but I think it would be good to revisit it in the future